### PR TITLE
webpack configuration setup

### PR DESCRIPTION
**Pull request by Velenir**
Important changes:
+ Code splitting
+ Sourcemaps for everything
+ Minification of everything
+ Compilation for `.scss` files enabled
+ Eslint, Stylelint and Tslint
+ TypeScript integration
+ Linting (with `--fix`) only staged files in `precommit`
+ Switched to React v16 and react-router v4

Future plans/poposals:
+ Storybook integration
+ Run tests in `prepush`

### CODE REVIEW

**Review by W3stside — APPROVED** [review]
Looks good and compiles for me. Fast re-compiling is a real nice add. Solid work.

File: .babelrc [hunk 1]
```diff
@@ -1,7 +1,12 @@
 {
   "presets": [
     "react",
-    "env",
+    [
+      "env",
+      {
+        "modules": false
```
Comment (apanizo) [inline .babelrc:7]: Can you add comments?

File: .babelrc [hunk 2]
```diff
@@ -12,8 +17,14 @@
   "env": {
     "production": {
       "plugins": [
-        "transform-react-remove-prop-types"
+        [
+          "transform-react-remove-prop-types",
+          {
+            "mode": "remove",
```
Comment (apanizo) [inline .babelrc:23]: Here as well. Any particular reason for doing it?

File: .babelrc (hunk 1 above)
Comment (Velenir) [reply .babelrc:7]: This disables  ES module `import` handling by babel (transpilation into `require`) and leaves it to webpack. Webpack uses `import` (especially named `import`) to perform Tree shaking.

File: .babelrc (hunk 2 above)
Comment (Velenir) [reply .babelrc:23]: `"mode": "remove"` together with `"removeImports": true` on the next line configures the plugin in addition to removing constructs `Component.propTypes={...}` to also remove `import PropTypes from 'prop-types'`. Shaves off another couple lines per Component.

File: webpack.prod.config.js [hunk 3]
```diff
@@ -1,9 +1,14 @@
+/* eslint-disable import/no-extraneous-dependencies */
 const ExtractTextPlugin = require('extract-text-webpack-plugin')
 const HtmlWebpackPlugin = require('html-webpack-plugin')
 const FaviconsWebpackPlugin = require('favicons-webpack-plugin')
+const NameAllModulesPlugin = require('name-all-modules-plugin')
+const BabiliPlugin = require('babili-webpack-plugin')
```
Comment (mmv08) [inline webpack.prod.config.js:6]: Babili has been renamed to babel-minify and they also released a new plugin for webpack:
https://www.npmjs.com/package/babel-minify-webpack-plugin

**Pull request closed without merging**